### PR TITLE
UrlSigning quick start sample

### PR DESCRIPTION
**Pull request by gball14**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                "name": "origin1",
+                "properties": {
+                  "hostName": "[parameters('originUrl')]"
+                }
+              }
+            ],
+			"urlSigningKeys": [
+				{
+				  "keyId": "key1",
+				  "keySourceParameters":{
```
Comment (bmoore-msft) [inline 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: Seems like these values should be provided by the user?

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: @bmoore-msft , Yes. these should ideally be provided by customer pointing to their keyvault. As the repro try to submit the template to Azure RP, we created a sample key vault and referenced here. Let me know if there are better way to handle this. Users who use the template will not know what they keys are as they don't have permissions. Only Azure CDN APP (Principal) will have access to this key vault.

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: @bmoore-msft , Thanks for the clear ask. I updated the readme file. Please take a look.

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: Hi @bmoore-msft , we are waiting to share this example in our user documentation. Can you please look into this PR?

File: 201-cdn-with-ruleseengine-urlsigning/README.md [hunk 2]
```diff
@@ -18,3 +18,10 @@
 
 This template creates a CDN Profile and a CDN Endpoint with a user specified origin and all of our most commonly used settings on CDN. This template also configures rules engine UrlSigning action for default and override parameters.
 
+Before using the template, make sure to do the following:
+
+1. Create a key vault in the same subscription or in a different subscription.
+2. Define two secrets and set up secret values manually (at least 32 size)
```
Comment (bmoore-msft) [inline 201-cdn-with-ruleseengine-urlsigning/README.md:24]: does the secret value matter?

File: 201-cdn-with-ruleseengine-urlsigning/README.md [hunk 3]
```diff
@@ -18,3 +18,10 @@
...
+3. References the two secrets in urlSigningKeys of the template (change subscriptionId, resourceGroupName, vaultName, secretName, secretVersion)
+4. Make sure the keyvault is given "Get Permissions for Secret" access to "Microsoft.Azure.CDN" Principal.
```
Comment (bmoore-msft) [inline 201-cdn-with-ruleseengine-urlsigning/README.md:26]: how do I do this?  the principal does not exist in the tenant...

File: 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: A couple comments on the readme - also, the values below need to be parameters for the template.

File: 201-cdn-with-ruleseengine-urlsigning/README.md (hunk 3 above)
Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:26]: @bmoore-msft , Please see below. 
image

File: 201-cdn-with-ruleseengine-urlsigning/README.md (hunk 2 above)
Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:24]: Min length is 32. We allow any value.

File: 201-cdn-with-ruleseengine-urlsigning/README.md (hunk 3 above)
Comment (bmoore-msft) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:26]: the principal does not exist in the tenant...

File: 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:86]: These still need to be moved to parameters

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/azuredeploy.json:128]: Parametrized the urlsigning keys.

File: 201-cdn-with-ruleseengine-urlsigning/README.md (hunk 3 above)
Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:26]: @bmoore-msft , what tenant are you using? I checked with both COPR and AME tenant. I could see the principal.

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:26]: You may need to register if it doesn't exist. https://docs.microsoft.com/en-us/azure/cdn/cdn-custom-ssl?tabs=option-2-enable-https-with-your-own-certificate

image

Comment (gball14) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:26]: @bmoore-msft , can you please check if all looks good for you?

Comment (bmoore-msft) [reply 201-cdn-with-ruleseengine-urlsigning/README.md:23]: let's add this step to the readme... so others don't get hung up.  It looks like the SP is given contributor perms to the sub by default so I suspect that step will always be required.

Everything else looks good.

**Pull request merged**